x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/IndicesAccessControl.java

```diff
@@ -21,11 +21,53 @@
  */
 public class IndicesAccessControl {
 
-    public static final IndicesAccessControl ALLOW_ALL = new IndicesAccessControl(true, Collections.emptyMap());
-    public static final IndicesAccessControl ALLOW_NO_INDICES = new IndicesAccessControl(true,
-            Collections.singletonMap(IndicesAndAliasesResolverField.NO_INDEX_PLACEHOLDER,
-                    new IndicesAccessControl.IndexAccessControl(true, new FieldPermissions(), DocumentPermissions.allowAll())));
-    public static final IndicesAccessControl DENIED = new IndicesAccessControl(false, Collections.emptyMap());
+    public static final IndicesAccessControl ALLOW_ALL = new IndicesAccessControl(true, null) {
+        @Override
+        public IndexAccessControl getIndexPermissions(String index) {
+            assert false == Regex.isSimpleMatchPattern(index) :
+                    "index access control can and should only be retrieved for a given concrete name, but requested [" + index + "]";
+            return IndexAccessControl.PROMISCUOUS_ACCESS_CONTROL;
+        }
+    };
+    public static final IndicesAccessControl ALLOW_NO_INDICES = new IndicesAccessControl(true, null) {
+        @Override
+        public IndexAccessControl getIndexPermissions(String index) {
+            for (String placeHolderIndexName : IndicesAndAliasesResolverField.NO_INDICES_OR_ALIASES_ARRAY) {
+                if (placeHolderIndexName.equals(index)) {
```

Comment:
Others will say a `Set` would be preferable to a `List`...
I don't know, I feel introducing a new public static variable with the same contents as an existing variable, to improve the usage code in a single place feels like a stretch to me. If you feel strongly about it I can introduce a `Set` or a `List`.

Reply:
But it's not introducting a new variable. The variable [already exists](https://github.com/elastic/elasticsearch/pull/59739/files#diff-cf5207b7cd7046339344922ccc0396cbL53). It got deleted with this PR. I was asking whether we should just leverage it instead of deleting it.

Reply:
`IndicesAndAliasesResolverField` now contains `public static final List<String> NO_INDICES_OR_ALIASES_LIST = Arrays.asList(NO_INDICES_OR_ALIASES_ARRAY);` so we can use the `contains` method here.